browser/ui/views/brave_actions/brave_shields_action_view.cc

```diff
@@ -215,6 +216,14 @@ void BraveShieldsActionView::OnTabStripModelChanged(
     TabStripModel* tab_strip_model,
     const TabStripModelChange& change,
     const TabStripSelectionChange& selection) {
+    
+  if (change.type() == TabStripModelChange::kRemoved) {
```

Comment:
I assume that this `kRemoved` noti will be sent after we receive active tab changed for that tab first.
Can you check this?

Reply:
I think noti will be like below?
* Closing non active tab - maybe `Removed` is only sent.
* Closing active tab - maybe we get active tab changing noti. and then `kRemoved` will be arrived.

Reply:
The flow is as below:

1. Non active tab: removed is only fired
2. Active tab: 
[27785:259:0406/172038.657028:ERROR:brave_shields_action_view.cc(221)] removed
[27785:259:0406/172038.668865:ERROR:brave_shields_action_view.cc(229)] active tab

Reply:
Ok, in that case `selection.old_contents` is null?

Reply:
`selection.old_contents` is the removed one. It's not null.

Reply:
Hmm, that means observering is started and ended properly with existing logic.

Reply:
maybe to fit with the BraveShieldsActionView lifecylce we should have a RemoveObserver in it's dtor or perhaps override `WillCloseTabs`?

Reply:
If `BraveShieldsActionView` is destroyed first before tab strip model handles all tabs active state while shutdown,
we should do call `RemoveObserver` for current active tab in dtor.

Reply:
`WillCloseAllTabs` is called first and then dtor of `BraveShieldsActionView`

Reply:
as discussed in DM this is an issue with the view closing before the active tab. We originally discussed doing this in the destructor and then I got off track and forgot about the original issue when we looked at the tab strip observer stuff. I think both changes are necessary.

Reply:
@simonhong I actually don't think it should be possible for the view to be destroyed before the webcontents because it's the button in the urlbar, but I think it's still best to add it to the dtor in case we're missing something in the tab strip observing